addons/html_editor/static/src/editor/core/collaboration_plugin.js

```diff
@@ -0,0 +1,324 @@
+import { HistoryPlugin } from "./history_plugin";
+
+// 60 seconds
+const HISTORY_SNAPSHOT_INTERVAL = 1000 * 60;
+// 10 seconds
+const HISTORY_SNAPSHOT_BUFFER_TIME = 1000 * 10;
+
+/**
+ * @typedef { Object } CollaborationPluginConfig
+ * @property { string } peerId
+ */
+
+export class CollaborationPlugin extends HistoryPlugin {
+    static name = "history_collaboration";
+    externalStepsBuffer = [];
+    // @todo @phoenix move this in the collaboration plugin ?
+    /** @type { CollaborationPluginConfig['peerId'] } */
+    peerId = null;
+
+    setup() {
+        this.DOMPurify = DOMPurify(this.editable.ownerDocument.defaultView);
+        this.peerId = this.config.plugins?.history_collaboration?.peerId;
+        this.serializable = typeof this.peerId !== "undefined";
+
+        if (this.config.plugins?.collaboration?.initialHistoryId) {
+            this.setInitialId(this.config.plugins.collaboration.initialHistoryId);
+        }
+
+        if (this.peerId) {
+            this._snapshotInterval = setInterval(() => {
+                this.makeSnapshot();
+            }, HISTORY_SNAPSHOT_INTERVAL);
+        }
+
+        super.setup();
+    }
+
+    /**
+     * @override
+     */
+    clean() {
+        super.clean();
+        this.branchStepIds = [];
+    }
+    processHistoryStep(step) {
+        step.peerId = this.peerId;
+        return step;
+    }
+    /**
+     * @override
+     */
+    reset() {
+        super.reset();
+        const firstStep = this.steps[0];
```

Comment:
what appends if you don't have any step ?

Reply:
`super.reset();` create the step